KarlDeux/magento2
# Added translate.test.js for Jasmine compatibility unit testing.

Fixes #12342 

Refactoring and creation of Jasmine Unit testing.
Removed JSTestDriver.



### Description

Currently Magento is using 2 different sets of tools for the JavaScript Testing:

JSTestDriver, which is considered legacy and is not supported anymore by the core team.
Jasmine, which was introduced to replace JSTestDriver. All new JS tests are implemented using it and executed in multiple CI environments, including public Travis CI and Magento's in-house CICD infrastructure.
To remove legacy framework it is required to reimplement remaining tests using Jasmine, and completely remove JSTestDriver support afterwards.

### Fixed Issues (if relevant)

1. magento/magento2#12342 : JSTestDriver removal

### Manual testing scenarios

1. Make sure that NPM is installed, if not `$ npm install`.
2. Generate static view files in Magento that are going to be tested `$ php bin/magento setup:static-content:deploy -f`.
3. Run Jasmine testing `$ grunt spec:`.

All the tests should pass.

### List of JS which require attention:

- [ ] [JsTestDriver/testsuite/mage/accordion/accordion.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/accordion/accordion.js)
- [ ] [JsTestDriver/testsuite/mage/collapsible/test-collapsible.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/collapsible/test-collapsible.js)
- [ ] [JsTestDriver/testsuite/mage/dropdown/test-dropdown.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/dropdown/test-dropdown.js)
- [ ] [JsTestDriver/testsuite/mage/edit_trigger/edit-trigger-test.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/edit_trigger/edit-trigger-test.js)
- [ ] [JsTestDriver/testsuite/mage/form/form-test.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/form/form-test.js)
- [ ] [JsTestDriver/testsuite/mage/menu/test-menu.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/menu/test-menu.js)
- [ ] [JsTestDriver/testsuite/mage/translate/translate-test.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/translate/translate-test.js)
- [ ] [JsTestDriver/testsuite/mage/translate_inline/translate-inline-test.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/translate_inline/translate-inline-test.js)
- [ ] [JsTestDriver/testsuite/mage/validation/test-validation.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/validation/test-validation.js)
- [ ] [JsTestDriver/testsuite/mage/decorate-test.js](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/decorate-test.js)
- [ ] [JsTestDriver/testsuite/mage/calendar](https://github.com/magento/magento2/tree/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/calendar)
- [ ] [JsTestDriver/testsuite/mage/loader](https://github.com/magento/magento2/tree/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/loader)
- [ ] [JsTestDriver/testsuite/mage/suggest](https://github.com/magento/magento2/tree/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/suggest)
- [ ] [JsTestDriver/testsuite/mage/tabs](https://github.com/magento/magento2/tree/2.2-develop/dev/tests/js/JsTestDriver/testsuite/mage/tabs) 

### Contribution checklist
